src/libmeasurement_kit/ooni/dns_injection.cpp

```diff
@@ -16,10 +16,14 @@ void dns_injection(std::string input, Settings options, Callback<Var<Entry>> cb,
     Var<Entry> entry(new Entry);
     (*entry)["injected"] = nullptr;
     templates::dns_query(entry, "A", "IN", input, options["backend"],
-                         [=](Error, dns::Message message) {
+                         [=](Error err, Var<dns::Message> message) {
                              logger->debug("dns_injection: got response");
-                             if (message.error_code == DNS_ERR_NONE) {
-                                 (*entry)["injected"] = true;
+                             if (!err) {
+                                 if (message->error_code == DNS_ERR_NONE) {
+                                     (*entry)["injected"] = true;
+                                 } else {
+                                    // This should never happen
+                                 }
```

Comment:
I'd rewrite the two if chains as

```C++
if (!err && message->error_code == DNS_ERR_NONE) {
    (*entry)["injected"] = true;
} else {
    (*entry)["injected"] = false;
}
```

This is basically what you have already written, given that you have an empty `else` branch.